core/domain/acl_decorators.py

```diff
@@ -68,6 +76,39 @@ def check_activity_accessible(
             activity_rights.viewable_if_private)
 
 
+def check_exploration_editable(user_id, user_actions, exploration_rights):
+    """Returns a boolean to signify whether given exploration is editable
+    by the user or not.
+
+    Args:
+        user_id: str. Id of the given user.
+        user_actions: list(str). List of actions given user can perform.
+        exploration_rights: Exploration rights object.
```

Comment:
The 500 exception can be caught and re-thrown as a user-visible exception (PageNotFound etc.). But the user-visible exceptions should not be used too far down into the code.

I see the issue with the helper function. I think passing in `self` is too much. Would it make sense to pull the exceptions that are currently in base.py out of it? Then you can pass the exception classes you need as args to the decorator.